kafka/consumer.go

```diff
@@ -187,39 +184,16 @@ func NewConsumerFromPartition(topic string, partition int, opts ...ClientOption)
 	return kafkaClient, nil
 }
 
-func routeMsg(c *KafkaConsumer, msgs <-chan *sarama.ConsumerMessage, rawMessages chan *sarama.ConsumerMessage) {
-	for m := range msgs {
-		rawMessages <- m
-		// ------ Prometheus metrics ---------
-		bytesProcessed.
-			WithLabelValues(m.Topic, "consumer", c.groupID, strconv.Itoa(int(m.Partition))).
-			Add(float64(len(m.Value)))
-		messagesProcessed.
-			WithLabelValues(m.Topic, "consumer", c.groupID, strconv.Itoa(int(m.Partition))).
-			Add(1)
-		offsetMetrics.
-			WithLabelValues(m.Topic, "consumer", c.groupID, strconv.Itoa(int(m.Partition))).
-			Set(float64(m.Offset))
-		// For testing: Log when got a message
-		prefixFromEngine := ""
-		fmt.Printf(prefixFromEngine+"Got a message save to queue: Topic: %s; groupID: %s; Partition: %d", m.Topic, c.groupID, m.Partition)
-
-		// Only auto mark offset when automark is enabled and it's a group consumer
-		if c.autoMark && c.groupConsumer != nil {
-			c.groupConsumer.MarkOffset(m, "")
-		}
-	}
-}
-
-func constructConsumerMsg(ctx context.Context, rawMessages chan *sarama.ConsumerMessage, messages chan messaging.Event, errors chan error) {
-ConsumerLoop:
+func routeMsg(ctx context.Context, c *KafkaConsumer, msgs <-chan *sarama.ConsumerMessage, messages chan messaging.Event, errors chan error) {
+	defer close(messages)
 	for {
 		select {
-		case m, ok := <-rawMessages:
+		case m, ok := <-msgs:
 			if !ok {
-				break ConsumerLoop
+				return
 			}
-			// lagMetrics.WithLabelValues(s.Topic, "consumer", s.Partition).Set(float64(s.Lag))
+			log.Printf("Obtained message: Topic: %s; groupID: %s; Partition: %d, Value: %s", m.Topic, c.groupID, m.Partition, string(m.Value))
+			c.groupConsumer.MarkOffset(m, "")
 			messages <- &event{
```

Comment:
a layperson question: if msgs gives us an err, the loop breaks and `messages` closes right away, events that are existed in this channel won't guarantee to be processed, right?